hwbench/environment/block_devices.py

```diff
@@ -128,3 +136,62 @@ def parse_version(self, stdout: bytes, _stderr: bytes) -> bytes:
     def dump(self):
         self.run()
         return self.data
+
+
+class Sdparm(External):
+    """Dumps based on External abstract class Sdparm tool information for a block device"""
+
+    def __init__(self, out_dir: pathlib.Path, block_device_name):
+        self.device_name = block_device_name
+        self.cmd_name = "sdparm"
+        self.out_dir = out_dir
+        self.data: dict[str, dict[str, Any]] = {}
+        self.version = b""
+
+    @property
+    def name(self) -> str:
+        return f"sdparm_{self.device_name}"
+
+    def run_cmd(self) -> list[str]:
+        return ["sdparm", "--all", f"/dev/{self.device_name}"]
+
+    def parse_cmd(self, stdout: bytes, stderr: bytes):
+        section_name = ""
+        metric_name = ""
+
+        if stdout:
+            lines = stdout.splitlines()
+
+            for line in lines:
+                str_line = str(line)
+                re_section = re.match(r"^(.*):$", str_line)
```

Comment:
would be cool to get a comment to explain what you're trying to parse here

Reply:
rewrote the thing